### DISCUSSION TITLE
Variable default values

### DISCUSSION THREAD

**Discussion by tpluscode** [Orange-OpenSource/hurl#1349] [Ideas]
The idea is to have parametrised hurl file where paremeters are optional

```
POST /search
Content-Type: application/xml

<application xmlns="http://www.zetcom.com/ria/ws/module/search" >
 <modules>
 <module name="Object">
 <search limit="{{LIMIT = 10}}" offset="{{OFFSET = 0}}" />
 </module>
 </modules>
</application>
```

I could call `hurl search.hurl` to use the default limit/offset equal 10 and 0 respectively.

And then I could `hurl --variable LIMIT=100 --variable OFFSET=100 search.hurl` to change the paging params

**Comment by tjbrockmeyer** [comment]
Super old thread, but I am interested in this idea as well.

Thinking initially of bash syntax,
```bash
${variable:-valueIfEmpty}
```
which is supported by things like docker compose for instance.

I think that the syntax for it is a sort of convention that many would understand, but also it's not self-explanatory, requires knowledge of bash or others who have implemented the syntax, etc. Something like a simple `=` sign is maybe nicer.

Something that always bugs me about the way that docker compose utilizes the defaults is that you have to override the value in every single location that it's used. It would be nice if, for hurl, a syntax could be found that could set the default for the variable in all locations of the file for the case when it is not assigned by the user.
